### ISSUE TITLE
multiprocessing issues on windows

### ISSUE THREAD

**Issue by alihaydaroglu** [AllenInstitute/deepinterpolation#68]
### Issue
Running `example_tiny_ephys_training.py` in the examples folder. Inference fails if `training_params['use_multiprocessing'] = True`. See the error in the attached image. Setting this to `False` works without errors.
![image](https://user-images.githubusercontent.com/15041006/141394989-3853a092-bf1a-4541-b060-67fd9635ee7c.png)

### System Information
Windows 10, Version 20H2 Build 19042. Intel i7-11700K (8 cores)
Error reproduced with the following CUDA/TF combinations:
1. CUDA 10.1, cuDNN 8.1, TF 2.2.2
2. CUDA 11.4, cuDNN 8.2, TF 2.7

**Comment by jeromelecoq** [comment]
this issue is related to conflict when calling a script with a __main__ 

I added the following to the FAQ: 
Question:When training or fine-tuning a network for 2d movies, I get the following error (or similar) ?:

    WARNING:tensorflow:multiprocessing can interact badly with TensorFlow, causing...proper idiom
    in the main module:

        if __name__ == '__main__':
            freeze_support()
            ...

    The "freeze_support()" line can be omitted if the program
    is not going to be frozen to produce an... " 
Answer: This error is related to this post : https://stackoverflow.com/questions/20222534/python-multiprocessing-on-windows-if-name-main For running scripts that trigger multi-processing, they need to be calling under a separate header as such:

    if __name__ == ‘__main__‘:
Place this line at the top of your script, below the import statement and before all your script code. Then indent your script code so that the code gets call under main. This prevents the computer to call an infinite number of threads upon activation of sub-threads. We corrected our example python script to fix this issue.
